src/context.rs

```diff
@@ -236,7 +237,48 @@ impl Context {
     pub fn emit_event(&self, event: EventType) {
         self.events.emit(Event {
             id: self.id,
-            typ: event,
+            typ: event.clone(),
+        });
+        let context = self.clone();
+        async_std::task::spawn(async move {
+            // TODO synchronously is prob. better
```

Comment:
Yes, just `block_on` every async task, otherwise logs may be reordered.

Reply:
Can we just do this - Doesn't `block_on()` block the current thread, which is an executor thread in this case, so if there is only one executor thread (which may happen on one-core machines, I think?), then the async stuff inside can't be executed since the only executor thread is blocked? Or is `block_on()` smart enough not to do this?

Reply:
if we log the timestamp outside of this and then in the webxdc sort from timestamp, or is it so fast that even then stuff can be in the wrong order?

Reply:
Hmmm, also the `test_migrate_account()` test is flaky, and the `task::spawn()` seems to be the reason for this

Reply:
@dignifiedquire said that using block_on could lead to deadlocks; generally both are probably kinda fine, block_on because database access is blocking anyway and spawn because in most cases the log should be understandable, and we have a timestamp. Also, we can switch back and forth later.

Reply:
There is one problem with `spawn()` that I had temporarily forgotten: It introduces very undeterministic behavior and could e.g. make exporting a backup fail because we are still writing to the database in the background (that's also why the test above was flaky).

This tips the balance in favor of `block_on()` again for my opinion. Note again that usually we are not doing anything async at all, and database access is blocking. A proper solution will require a background loop where we send stuff over a channel.

We can still easily switch later.